packages/react-reconciler/src/ReactStrictModeWarnings.js

```diff
@@ -110,14 +111,19 @@ if (__DEV__) {
   const getStrictRoot = (fiber: Fiber): Fiber => {
     let maybeStrictRoot = null;
 
-    while (fiber !== null) {
-      if (fiber.mode & StrictMode) {
-        maybeStrictRoot = fiber;
+    let node = fiber;
+    while (node !== null) {
+      if (node.mode & StrictMode) {
+        maybeStrictRoot = node;
       }
-
-      fiber = fiber.return;
+      node = node.return;
     }
 
+    invariant(
+      maybeStrictRoot !== null,
+      'Expected to find a StrictMode component in a strict mode tree. ' +
+        'This error is likely caused by a bug in React. Please file an issue.',
```

Comment:
We only call `record` method (which calls this one) in strict trees. So it's an underlying assumption that we can find it.